tests/Types/TypeRegistryTest.php

```diff
@@ -102,6 +102,16 @@ public function testRegisterWithAlreadyRegisteredInstance(): void
         );
     }
 
+    public function testConstructorWithDuplicateInstance(): void
+    {
+        self::markTestSkipped('The necessary check has been removed in https://github.com/doctrine/dbal/pull/5036.');
+
+        $newType = new TextType();
+
+        $this->expectException(Exception::class);
+        new TypeRegistry(['a' => $newType, 'b' => $newType]);
+    }
```

Comment:
@greg0ire I need input here. This new test made sense in 3.7.x, but apparently we allow duplicates in the `instances` array since #5036. Shall I remove the test?